src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -537,7 +536,7 @@ def get_record_data(
         for key, variable in state.variables.items():
             try:
                 submission_value = dynamic_values[key]
-            except PathAccessError:
+            except KeyError:
```

Comment:
Was this changed to KeyError because the variable keys cannot have `.` in them?

Reply:
no, before it would implicitly rely on `glom` being used internally, which raises `PathAccessError` if you access the key (without checking that it's present). So that was a leaky abstraction.

However, you highlight a bug - this only applies for keys without period, I also need to make sure these path access errors get transformed into a proper KeyError when using dotted keys.

Reply:
heh, it already works correctly because `PathAccessError` subclasses `KeyError`... I've added a test and made things more explicit.